htlcswitch/hop/iterator.go

```diff
@@ -127,6 +128,10 @@ type BlindingProcessor interface {
 	// ephemeral key provided.
 	DecryptBlindedHopData(ephemPub *btcec.PublicKey,
 		encryptedData []byte) ([]byte, error)
+
+	// NextEphemeral returns the next hop's ephemeral key, calculated
+	// from the current ephemeral key provided.
+	NextEphemeral(*btcec.PublicKey) (*btcec.PublicKey, error)
```

Comment:
So now the processor is intended to be stateful? As IIUC, you need the new blinding factor which is only available after you processor the current blinded path. Still getting through the commits, but not sure we have the right abstraction here.